libs/wingc/src/type_check.rs

```diff
@@ -1376,6 +1392,31 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn check_json_serializable_or_validate_type(
+		&mut self,
+		actual_type: TypeRef,
+		expected_type: TypeRef,
+		exp: &Expr,
+		context: &TypeCheckerContext,
+	) -> TypeRef {
+		// Skip validate if in Json
+		if context.in_json {
+			if !actual_type.is_serializable() {
+				self.expr_error(
+					exp,
+					format!(
+						"Expected \"Json\" elements to be serializable, but got \"{}\" which is not serializable",
+						actual_type
+					),
+				);
+			}
+
+			return actual_type;
```

Comment:
We can get rid of a level of nesting with this re-org. Score us some never-nesting points with @Chriscbr :grin:
```suggestion
		if !context.in_json {
			return self.validate_type(actual_type, expected_type, exp);
		}

		if !actual_type.is_serializable() {
			self.expr_error(
				exp,
				format!(
					"Expected \"Json\" elements to be serializable, but got \"{}\" which is not serializable",
					actual_type
				),
			);
		}
		actual_type
```